### PR TITLE
add "npm run serve" script

### PR DESCRIPTION
**Pull request by callumlocke**
I've been using this small addition and finding it very convenient, thought you might like to consider it for the WSK to help people who edit their gulpfile a lot...

This adds `npm run serve`, an optional wrapper around `gulp serve`. It restarts `gulp serve` whenever you edit your gulpfile, or jshintrc, or package.json (all files for which modifications usually warrant a restart).

The downside is it's an extra dep. But it's only nodemon, which is a fairly light dep with [no postinstall](https://github.com/remy/nodemon/blob/master/package.json).

### CODE REVIEW

**Comment by callumlocke** [comment]
I signed it

**Comment by addyosmani** [comment]
Thanks a lot for this addition, @callumlocke. I've played around with `npm run` scripts a bunch for a few projects and agree, they are super useful. Unfortunately, I've also found them to be unreliable with npm@2 on Windows. I'll keep this open for now and see if we can get some further testing done before making a call.

**Comment by sindresorhus** [comment]
It think it would be confusing to have both `npm run serve` and `gulp serve` with different behaviour. I'd rather see this integrated in the `gulp serve` task.

**Comment by callumlocke** [comment]
I'm not sure how it could be integrated in the gulp serve task... I think the only way it can happen is with an extra process (eg nodemon) wrapping the gulp process. (Unless there's some way a single process can kill and restart itself?)

I agree it's a little confusing to have two commands doing almost the same thing... But I think it could be documented clearly in docs/commands.md, as an 'advanced' wrapper command around gulp serve, for people who edit their gulpfile a lot.

Or maybe this is 'recipe' territory?

**Comment by callumlocke** [comment]
@sindresorhus what if it was `npm start`, not `npm run serve`?

File: package.json [hunk 1]
```diff
@@ -34,6 +35,7 @@
   },
   "private": true,
   "scripts": {
+    "serve": "nodemon --watch gulpfile.js --watch package.json --watch .jshintrc --exec gulp serve",
```
Comment (sindresorhus) [inline package.json]: `gulp serve` must be quoted (single-quotes).

**Comment by sindresorhus** [comment]
`npm start` is better. I wouldn't personally use this, so would go with a recipe, but leaving it up to @addyosmani to decide.

**Comment by addyosmani** [comment]
We've made a call on not moving ahead with this. My view is in line with this comment from Sindre:

> It think it would be confusing to have both npm run serve and gulp serve with different behaviour. I'd rather see this integrated in the gulp serve task.

**Pull request closed without merging**